cpu/atmega_common/include/sys/types.h

```diff
@@ -7,22 +7,49 @@
  * directory for more details.
  */
 
- #ifndef AVR_TYPES_H
- #define AVR_TYPES_H
+/* If building on native we need to use the system libraries instead */
+#ifdef CPU_NATIVE
+#pragma GCC system_header
+/* without the GCC pragma above #include_next will trigger a pedantic error */
+#include_next <sys/types.h>
+#else
+#ifndef SYS_TYPES_H_
+#define SYS_TYPES_H_
 
-#include <inttypes.h>
+#include <stdint.h>
 #include <stddef.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-typedef int16_t suseconds_t;
```

Comment:
it's probably a good idea to move the proposed atmega changes to a seperate PR, as they are mostly unrelated to this new feature. (here and above)